lib/blockscore/collection.rb

```diff
@@ -1,18 +1,91 @@
 module BlockScore
   class Collection < Array
-    def initialize(target)
-      super()
-      @target = target
+    attr_reader :default_params
+    protected :default_params
+
+    attr_reader :parent, :target, :data
+
+    def initialize(params)
+      @parent = params.fetch :parent
+      @target = params.fetch :target
+      register_parent_data
+    end
+
+    def default_params
+      {
+        :"#{parent_name}_id" => parent_id
+      }
+    end
+
+    def parent_id
+      parent.id if parent.respond_to?(:id)
```

Comment:
The issue is that

```ruby
person = BlockScore::Person.new
person.id 
#=> NoMethodError: undefined method `id' for #<BlockScore::Person:0x3fded1db262c JSON:{
}>
```

because it checks the method_missing in `Base` checks attributes then fails, and I wanted to allow someone to be able to build a `Person`  and work with the collection still.